### DISCUSSION TITLE
& operator should support obtaining the address of static anonymous lambda methods

### DISCUSSION THREAD

**Discussion by bysdxt** [dotnet/csharplang#6111] [General] [answered]
```C#
unsafe
{
    static void h() { }
    delegate*<void> p = default;
    p = &h; //  this is ok
    p = &(static () => { });    // this should be ok too
}
```

**Comment by svick** [comment]
To me, this looks like something that would make function pointers slightly easier to use, with a somewhat peculiar syntax (because of the required parentheses). But function pointers are an advanced feature, which means small improvements in their convenience are probably not worthwhile.

**Comment by HaloFour** [comment]
I think that would also tie the compiler to a specific implementation of how static lambdas are emitted.  They currently are instance methods on a singleton instance, but that would be incompatible with the function pointer.

**Comment by 333fred** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/issues/3476.
